CoreAlgorithmPlugins/src/au/gov/asd/tac/constellation/plugins/algorithms/sna/centrality/PagerankCentralityPlugin.java

```diff
@@ -95,118 +92,181 @@ public PluginParameters createParameters() {
 
         return parameters;
     }
-
+ 
     @Override
     public void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException {
-        final boolean treatUndirectedBidirectional = parameters.getBooleanValue(TREAT_UNDIRECTED_BIDIRECTIONAL_PARAMETER_ID);
         final float dampingFactor = parameters.getFloatValue(DAMPING_FACTOR_PARAMETER_ID);
         final int iterations = parameters.getIntegerValue(ITERATIONS_PARAMETER_ID);
         final float epsilon = parameters.getFloatValue(EPSILON_PARAMETER_ID);
         final boolean normaliseByAvailable = parameters.getBooleanValue(NORMALISE_AVAILABLE_PARAMETER_ID);
 
-        // identify incoming connections and store connected vertices and their outgoing connection count
-        final Set<Integer> verticesWithZeroOutLinks = new HashSet<>();
-        final Map<Integer, Set<Tuple<Integer, Integer>>> vertexInLinks = new HashMap<>();
-        final int vertexCount = graph.getVertexCount();
-        for (int vertexPosition = 0; vertexPosition < vertexCount; vertexPosition++) {
-            final int vertexId = graph.getVertex(vertexPosition);
-            vertexInLinks.put(vertexId, new HashSet<>());
 
-            if (graph.getVertexEdgeCount(vertexId, GraphConstants.OUTGOING) == 0) {
-                verticesWithZeroOutLinks.add(vertexId);
+        PagerankVertex.initialiseAllPagerankVertices(graph, parameters.getBooleanValue(TREAT_UNDIRECTED_BIDIRECTIONAL_PARAMETER_ID), dampingFactor, normaliseByAvailable);
+        
+        // calculate pageranks
+        for (int iteration = 0; iteration < iterations; iteration++) {
+            interaction.setProgress(iteration, iterations, "Iteration " + iteration + " of " + iterations, true);
```

Comment:
This is my favourite line of code, i read it as "iteration, iteration, iteration, iteration, iteration.... true".

Reply:
I have named this line:
The iteration of the alliteration of iteration

But also i've renamed them to maxIterations and currentIteration